pkg/kubestatemetrics/store/store_test.go

```diff
@@ -398,6 +398,75 @@ func TestPush(t *testing.T) {
 	}
 }
 
+func TestReplace(t *testing.T) {
+	idsToAdd := map[string]string{"bec19172-8abf-11ea-8546-42010a80022c": "gke-charly-default-pool-6948dc89-g54n", "8b136387-8a51-11ea-8546-42010a80022c": "gke-charly-default-pool-6948dc89-4r7"}
+	creationTs := int64(709655400000)
+	storeName := "*v1.Node"
+	metricName := "kube_node_created"
+
+	genFunc := func(obj interface{}) []metric.FamilyInterface {
+		o, err := meta.Accessor(obj)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		metricFamily := metric.Family{
+			Name: metricName,
+			Metrics: []*metric.Metric{
+				{
+					LabelKeys:   []string{"node"},
+					LabelValues: []string{string(o.GetUID()), o.GetName()},
+					Value:       float64(o.GetCreationTimestamp().Unix()),
+				},
+			},
+		}
+		return []metric.FamilyInterface{&metricFamily}
+	}
+
+	ms := NewMetricsStore(genFunc, storeName)
+	for uid, name := range idsToAdd {
+		s := v1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              name,
+				UID:               types.UID(uid),
+				CreationTimestamp: metav1.Unix(creationTs, 0),
+			},
+		}
+		err := ms.Add(&s)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	ms.mutex.RLock()
+	metrics := ms.metrics
+	ms.mutex.RUnlock()
+
+	// Verify that our test is setup correctly and that both metrics are in place
+	assert.Contains(t, metrics, types.UID("bec19172-8abf-11ea-8546-42010a80022c"))
```

Comment:
I think it would be better to use `require` instead of `assert` in these two. If the test was not set up correctly, we shouldn't continue.